include/fastdds/dds/xtypes/dynamic_types/DynamicTypeMember.hpp

```diff
@@ -0,0 +1,139 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FASTDDS_DDS_XTYPES_DYNAMIC_TYPES_DYNAMIC_TYPE_MEMBER_HPP
+#define FASTDDS_DDS_XTYPES_DYNAMIC_TYPES_DYNAMIC_TYPE_MEMBER_HPP
+
+#include <memory>
+
+#include <fastdds/dds/core/ReturnCode.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/AnnotationDescriptor.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/MemberDescriptor.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/VerbatimTextDescriptor.hpp>
+#include <fastrtps/fastrtps_dll.h>
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+
+class DynamicTypeMember;
+
+template<>
+struct traits<DynamicTypeMember> : public object_traits<DynamicTypeMember>
+{
+    using base_type = DynamicTypeMember;
+};
+
+/*!
+ * Represents a "member" of a type. A "member" in this sense may be a member of an aggregated type, a constant within an
+ * enumeration, or some other type substructure.
+ */
+class RTPS_DllAPI DynamicTypeMember : public std::enable_shared_from_this<DynamicTypeMember>
+{
+public:
+
+    using _ref_type = typename traits<DynamicTypeMember>::ref_type;
+
+    /*!
+     * Provides a summary of the state of this type overwriting a provided object (see [standard] 7.5.2.6.2)
+     * @param[inout] descriptor @ref MemberDescriptor reference where copied the information.
+     * @return standard @ref ReturnCode_t
+     */
+    virtual ReturnCode_t get_descriptor(
+            traits<MemberDescriptor>::ref_type md) = 0;
```

Comment:
```suggestion
            traits<MemberDescriptor>::ref_type descriptor) = 0;
```

`md` seems like Barro's inheritance :laughing:

Reply:
Done in 1596a8a54